docs/src/mock.md

```diff
@@ -41,31 +41,99 @@ page.route("https://dog.ceo/api/breeds/list/all", handle)
 ```csharp
 await page.RouteAsync("https://dog.ceo/api/breeds/list/all", async route =>
 {
-    var json = /* JsonElement with the test payload */;
+    var json = new { message = new { playwright_dog = new List<string>() } };
     await route.FulfillAsync(new () { Json: json });
 });
 ```
 
 ```java
 page.route("https://dog.ceo/api/breeds/list/all", route -> {
   route.fulfill(new Route.FulfillOptions()
-    .setBody("{\"message\":{\"test_breed\":[]}}"));
+    .setBody("{\"message\":{\"playwright_dog\":[]}}"));
+});
+```
+
+In the example below we intercept the route to the dog API and return a custom response. We then go to that url and assert that the response is our mock data:
+
+```js
+test("mocks the dog api", async ({ page }) => {
+  await page.route("https://dog.ceo/api/breeds/list/all", async (route) => {
+    const json = {
+      message: { playwright_dog: [] },
+    };
+    await route.fulfill({ json });
+  });
+  await page.goto("https://dog.ceo/api/breeds/list/all");
```

Comment:
How can we `goto` to the url that is a json api call? The page url must be different from the api url.

Reply:
because we dont have a real world example to use that has a frontend that they can see the results I thought it was a good idea to use the api url to go to as they can then see the changes direct in the json. Basically if a user copies that code snippet and runs it it will work exactly as it does which is really nice and easy for the user to see and understand. I find it super useful to visually see what is going on but open to suggestions

Reply:
I understand the motivation. But I think it turns a real example into useless one. Now, as a user, I would think that I can only mock the url that I directly `goto` to, which is never the case. In practice, I never mock the html, so this is very misleading imo. Should we find another example then?